Ballistics Firearm Operations Overhaul and New Improvised Firearms

VickiMorris opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<details>
<summary>To-Do List for PR; Final Stages</summary>
Crossed-out checkboxes are ideas that were initially part of the PR, but have since been made defunct by changes over the month this has been written.

- [x] ~~Add .41 cal ammo types~~
- [x] Implement Pipe Rifle & Pistol
- [x] Sprites for Lever Rifle, Pipe guns, Ammo type
- [x] Craftable Med-Capacity Semi-Auto Gun / Bootleg WT-550
- [x] ~~Proper Muzzle Loading Musket / Bootleg Sniper~~
- [x] Sprites for newly mentioned firearms
- [x] **In-hand sprites** for everything
- [x] Bolt overlays for all ~~newly defined NB_BREAK and~~ PUMP action guns, to visually show state of the gun.
- [x] Crafting recipes
- [x] Refactor rifle loading code
- [x] ~~Modify NO_BOLT (break action guns) code to function similar to Mosins for loading/unloading~~
- [x] Modify Pump Action guns to function similar to the Mosin for cycling cartridges
- [x] Clean up and streamline code.
- [x] Balance balance balance ammo and firearms (Might take a bit)

</details>

The overall goal of this PR is to make the aquisition of firearms and ammunition by antagonists (and by extension, crew and security in dire circumstances) easier, by providing more (lower quality) firearms and ammunition to be crafted. Paired with this is a series of tweaks to existing firearms to change, what I feel, are oversights or just poor design choices. This is the third re-write of this PR about section, so I will keep it to a bulleted list and not turn it into a 'used-car salesman' pitch.

---
- Two new firearms, a single shot break-action rifle chambered in 7.62 (same ammo type as the Mosin), and an open-bolt semi-auto 'carbine' chambered in 9mm (used by the Stechkin APS, Saber SMG, and Uzi).
- A lever action rifle; 8+1 'pump'-action gun chambered in .38, currently unobtainable and a byproduct of previous testing.
- Makeshift ammo types for 7.62, 9mm, 10mm, .38, .357, and a new type of shotgun ammo.
- Improvised shotgun ammo now has better range; each pellet has a random range of '3 to 8' instead of '1 to 8' 
- ~~The improvised shotgun can no longer chamber 'high-power' rounds; this includes buckshot, lethal slugs, and all tech shells asides from dragon's breath. All other ammo still function normally.~~
- The improvised shotgun has a 10% chance to explode when firing 'high-power' rounds (Buckshot, Slugs, and all Tech Shells asides from dragons breath), increasing by 5% each time a high-power round is fired from it. After suffering enough 'barrel stress' (30 stress / 6 shots) from firing high-power rounds, there is a 5% chance that low-power rounds will also cause a backfire. An additional variable has been included to allow admins and mappers to create improvised shotguns that ignore this mechanic.
- Slightly more complex crafting recipes for all improvised guns, requiring a couple extra components and a welder to construct.
- Improvised weapons are no longer loaded when crafted, requiring ammo to be made separately. They no longer require or spawn with a removable firing pin either.
- Refactoring of the Mosin rifle code, allowing it to be applied to other guns. The Sniper now uses this bolt type and fires marginally slower as a result.
- The ability to half-cock 'PUMP' bolt weapons (most shotguns). The ability to directly chamber rounds in PUMP and TWO_STEP bolt weapons while the bolt is locked open. This applies to most shotguns, the Mosin, and the Sniper Rifle.
- Hacksaws, obtainable from hacked autolathes; it can be used as a slower working and lower damage substitute to medical circular saws. This includes surgery, sawing weapons, and butchering.
- Sawing weapons actually increases the recoil of guns by 1 now, instead of setting the recoil to 1. Almost all sawn-off guns have higher recoil as a result.
- Replacing spent ammunition in a magazine has been entirely gutted. Magazines loaded with spent ammo must be emptied before they can be reloaded; this mainly effects NO_BOLT type guns (Break-Action Shotguns and Revolvers)
- Ammo boxes and removable magazines are all larger, now size category SMALL instead of TINY. This is to cut down on the *absurd* amount of ammo that can be carried for magazine fed guns.
- Removable magazines are no-longer filled instantly from ammo boxes, loaded at a rate of 1 bullet per 4 deciseconds. This can be done while moving, if reloading magazines on the go is needed. They can no longer pick up bullets from the floor like regular ammo boxes can either.
- Guns no longer chamber a round during their initialization, and must be manually cycled before firing for the first time. They still spawn with the same amount of ammo. An exception is made for NO BOLT guns, as there is no way to do this without giving them less ammo.
- The admin-only trait "NICE_SHOT" has been expanded upon. In addition to granting a wider ricochet and auto-aim degree, it allows the user to rack PUMP and TWO_STEP firearms with a single hand. This will likely never see use, but allows the creation of a supernaturally strong gunslinger through admin shenanigans.

<details>
<summary>For the nitty-gritty statistics, please click here.</summary>

- Pipe Carbine
Bulky-Sized firearm, can be stored on Belt slot only. Has a fire rate of 2 bullets per second, and an innate spread of 10 degrees. Chambered in 9mm (20 base damage).
The craftable magazines can hold 9 bullets, and can be upgraded with matter bins; each tier lets it hold 3 additional bullets, upto a max of 18 bullets with a bluespace matter bin.
- Pipe Rifle
Bulky-Sized firearm, no innate item slot. Can have a sling made from 10 cable coil to allow it to be carried on your Back and can be sawn-off, similar to the improvised shotgun. Chambered in 7.62 (40 base damage).

New Ammo Types
---
```
- 9mm (Improv); 20 damage, 10 innate spread
- 10mm (improv); 27 damage, 10 innate spread
- .38 (Improv); 25 damage, 1 ricochet (no auto-aim), 10 innate spread
- .357 (Improv); 50 damage, 15 innate spread, -10 AP
- 7.62 (Improv); 27-30 damage (randomized), 0 AP
- 7.62 (Hotload); 32-35 damage, 15 AP, '0.7 speed' (decisecond per tile, base is 0.8)
- Shotgun (Glasspack); 5 pellets, 15 degree cone. 6 damage per pellet (30 point-blank)
Max range of 8 tiles, 0.5 damage drop-off per pellet per tile.
80% chance for each pellet to 'shrapnel' and embed (60% embed chance).
```

</details>

---
Sprites for the lever action rifle were derived from the original lever action shotgun sprite, as made by OceanFish1 for #3153
Sawing noise as used for the hacksaw is derived from https://freesound.org/people/larryblag/sounds/333288/ by larryblag, originally released under Creative Commons 0.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The second half of this PR is easy enough to justify, the options for obtaining firearms as an antag at the moment are to order it from cargo (essentially steal it from security), actually steal it from security, purchase it from an uplink (BBs, Heretics, Changelings, Cult, etc stay losing), get Gamer Loot as a shaft miner or explorer, or craft the single improvised gun that's available. Having more options for antags to work their way towards better gear is good, especially when ranged options for most are pretty weak.

This doesn't directly change how Standard, Locking, and Open bolt guns function, but indirectly tweaks them with some changes to how magazines function; a bulk of this overhaul looks at changing the Pump bolt type to allow half-cocking and directly chambering round (functioning as a half-automatic version of the Mosin's bolt type), and defining a new bolt type (and reclassifying a couple of firearms in the process).

This is to make certain logical choices more favorable; having a gun with an larger internal magazine is better because I can fire more before needing to reload (Improv Shotgun is weaker than Bartender's Shotgun is weaker than Combat Shotgun) and reloading takes time. Having multiple magazines for my gun means I have more ready-to-use firepower in exchange for actual ammo capacity (Carrying multiple loaded magazines vs. one magazine and multiple boxes of ammo).

The slight logic changes to TWO_STEP and PUMP firearms allows more tactical usage of them; being able to chamber rounds directly is a buff to shotguns and the sniper, as you can now choose your next shot without needing to mess around with loading ammo into the magazine and then racking to charge the gun. While this isn't as much of an issue with shotguns, I found it miserable to actually load any of the specialty ammo into the sniper rifle without having a spare mag for the two bullets you get in each specialty ammo box.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>
The lever action rifle and pipe pistol are both derivative sprites from the lever-action shotgun and improvised shotgun, respectively. The ammo box fills out the boxes of ammo with the 4th color (others being red, blue, and green), the actual .41 caliber cartridges are based off of the Mosin / 7.62mm cartridge, and the improvised ones looking like a slimmed down improvised shotgun shell. Since removing the .41 caliber cartridge, the lever rifle is now chambered in .38 (same as the Det's revolver) and the new ammo box and proper metal cartridge sprites have been removed.

The original lever-action shotgun and improvised shotgun are off to the right (on the white tiles), and all the new sprites are on a blue background for better contrast.
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/c0b901cb-9e9c-4cef-8774-9376721e2853)

The pipe SMG / pipe repeater now exists! A slightly more expensive to make full auto (but slow firing) open bolt mag fed firearm (magazine not included in crafting). It uses the *very powerful* 9mm cartridge (at 20 damage per shot), has mild spread, and an upgradable magazine (initially stores 9 rounds, can be upgraded to 18 with a T4 matter bin)
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/50d91553-bb04-4e9d-8a32-6c521d03482d)

General screenshot showing the remaining new items, and some of the new bolt overlays used for shotguns to represent being locked open (to make it easier to recognize at a glance)
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/5d4d2b72-8f44-4d46-808c-d7a15965289a)

New inhand sprites, used for the improvised shotgun and pipe rifle (and their respective sawn-off variants )
![custominhands](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/fd4d6015-f4bc-4aed-bd3e-4e8f1dfe7014)

Recipes as of time of marking this as ready for review
![gun recepies](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/794dfaed-47ab-45bb-a659-edf97a29b478)
![ammo recepies](https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/7fd7eb15-550a-4129-8681-b93a40c89cde)

Videos below document some of the latest changes, and guns in their current state in the PR.

https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/49b92c3a-6a6f-4e70-8d7b-d0fbc593212a


https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/8a1b08fe-9358-484d-9818-473c31a20cc7

This video documents the changes to the improvised shotgun, and detonation/barrel stress mechanic while firing high-power rounds.

https://github.com/BeeStation/BeeStation-Hornet/assets/97719613/4bf38da9-b2b7-4d08-b9e8-def64dac9623

</details>

## Changelog
:cl: Impish_Delights, OceanFish1
add: New craftable a7.62, shotgun, 9mm, 10mm, .38, and .357 ammo types. Lever action rifle, craftable pipe rifle and pipe pistol, craftable pipe SMG and magazines.
add: Two-step bolt guns (Sniper & Mosin) can have ammo directly chambered, allowing loading a single shot without a magazine.
add: Pump-action guns can be half-cocked, locking them open to allow directly chambering rounds as well.
add: Hacksaws are now available from hacked autolathes, very slow manual saw for surgery and cutting down guns. The Russian Smuggler hunter shuttle spawns with one as well.
soundadd: Sawing noise for use with new hacksaw item. Original sound by larryblag on freesound.org
imageadd: Sprites for new ammo types, guns and accessories. Custom in-hand sprites for improv shotgun and pipe rifle. Original lever action shotgun sprite made by OceanFish1.
balance: Improvised shotguns have an incremental chance to explode when firing high-power ammo (buckshot, slugs, most tech shells). No danger with improv ammo and all other ammo types (incen, dart, rubbershot, beanbag, improv, breacher, dragonsbreath).
balance: The Sniper Rifle now uses the same bolt type as the Mosin Nagant. Fire rate is marginally slower as a result, and requires two interactions to cycle a new cartridge.
balance: Sawing adds 1 recoil instead of setting recoil to 1. Sawn-off shotguns have higher recoil across the board as a result. Slightly lowers Improv shotgun's base recoil as it would be absurd otherwise.
balance: The derringer has weapon weight 0.5 instead of 1, and no longer has hard-defined unwielded spread to bring it in line with the changes from high speed gun accuracy
balance: Improvised guns no longer require firing pins, and longer give free ammo when crafted.
balance: Improvised shotgun shells have better minimum range.
balance: All ammo boxes and external magazines are size SMALL instead of size TINY now. Magazines now load 1 bullet at a time, and no longer vacuum up bullets.
tweak: Due to rising safety concerns, guns are no longer shipped with a chambered round, and must be manually cycled before use.
tweak: Improvised weapons require an igniter, and a welder to construct now.
tweak: The admin-only trait "NICE_SHOT" now allows the user to cycle PUMP and TWO_STEP bolt firearms with a single hand, instead of requiring both.
refactor: Changes mosin/boltaction rifle code to be a new bolt type instead of bespoke code for the rifle parent type.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

VickiMorris committed:
Adds .41 caliber, lever rifle

Adds .41 caliber lever action rifle, 7+1 shot "pump action"
Adds .41 caliber ammo, 35 damage base, slightly faster travel time
.41 paper cartridge, 30 damage
.41 hot load paper, 30 damage, even faster travel, mild armor pen

VickiMorris committed:
Piperifle, Recoil, .41 Ammo Box

Adds the pipe rifle as an item
Adds recoil to the pipe rifle and lever action rifle
Adds a unique sprite to the .41 cal ammo box (yellow :D)
Adjusts papershot damage

VickiMorris committed:
Adds TWO_STEP bolt type, refactors bolt rifle code

Adds new TWO_STEP gun bolt type (Like pump, but toggles between locked and unlocked bolt on interaction, and can only be loaded with a locked/open bolt)
Refactors Mosin/Bolt Rifle racking code to be included with the parent ballistic.dm file.
.41 Caliber Paper cartridges have mildly randomized damage now.

VickiMorris committed:
FIXES A LOADING FUCKUP

Hehe put an OR instead of an AND on checking when loading ammo.

VickiMorris committed:
New Sprites for Lever Rifle

New sprite for lever action rifle
Removes firing pin from lever action rifle
further tweaks projectile speed of .41 cal ammo

VickiMorris committed:
Proper pipe gun sprites

Sprites for Pipe Rifle, Pipe Pistol (sawn off pipe rifle), and Pipe Gun Bolt
Makes sawing check the currently chambered round instead of the ENTIRE MAGAZINE before shooting your face

VickiMorris committed:
Merge branch 'master' into pipeGuns

VickiMorris committed:
Tweaks ballistics.dm & recoil, merges upstream

Merges upstream to get benefits of #10008 and tweaks to Mosin & sawing code to ensure compatibility.
Reduces base recoil of improv shotgun to 1.5 (from 2)
Tweaks sawing code so it ADDS 1 recoil instead of setting recoil to 1 (Mainly affects shotguns)
Tweaks Obrez code to take this into account for 1-handed firing
Removes vestigial code from rifle.dm as TWO_STEP bolt type has been fully implemented

VickiMorris committed:
Squashes sprite sheets

Didn't realize the merge conflict was over the sprite ballistic.dmi file, whoops! Merges the two sprite sheets so everything is nice textured.

Fixes a few other minor things with sawing, including a runtime with the new check on the chambered round

VickiMorris committed:
Ups .41 cal damage, improv shotgun nerf

Ups .41 caliber damage across the board, adds hollowpoint copper slugs.
Nerfs improvised shotgun, can no longer utilize standard lethals or tech shells

VickiMorris committed:
Merge branch 'master' into pipeGuns

VickiMorris committed:
Ports slow break action guns

Ports changes from "accurateBreakactionGuns" manually; NO_BOLT guns toggle between open and closed with ctrl-click, can only be loaded/unloaded when open and fired when closed.

VickiMorris committed:
Massively cleans up NO_BOLT cycling, new buckshot

Changes from Ctrl+Click to cycle to a three-step cycle for NO_BOLT guns, (Open -> Unload/Load -> Close).
Completely removes replace_spent on all magazines and ammo boxes
Converts the pipegun from a TWO_STEP to a NO_BOLT
Adds new 'Single-Aught Buckshot' - More pellets, less damage per pellet (8 @ 5.5 ea | 44 PB)

VickiMorris committed:
Shotguns pump like Mosins bolt

Splits shotgun racking into an explicit rack open / rack closed
Speeds up racking across the board to 3 (from 5) as it's a two-step process now (slightly slower at 6 total)
Keeps rifle rackspeed at 5 (Mosins stay losing)
Splits shotgunpump.ogg into two new files; shotgunpump_open.ogg and shotgunpump_close.ogg

VickiMorris committed:
Merge pull request #1 from VickiMorris/master

Port latest upstream (automatic shotgun changes)

VickiMorris committed:
Sniper changes, Magazine tweaks, Multiloading

Sniper rifle is now BOLT_TYPE_TWO_STEP and supports loading into chamber, and loading the attached magazine while the bolt is open.
Magazines no longer hoover up bullets like ammo boxes do.
Magazines without multi-load will load bullets one at a time when loading from an ammo box (1 bullet every 4 deciseconds)
Magazines with multi-load bypass this do_after for loading and load instantly (All internal magazines except shotgun)
Ballistic guns now have an explicit caliber variable, set to match the magazine's caliber on Initialization
Ballistic guns have the new 'direct_loading' variable, used for guns that support loading a removable magazine that's inserted, or loading into chamber when they otherwise wouldn't.
Tweaks to Det's revolver to work with new caliber changes
Guns no longer start with a chambered round (asides from NO_BOLT guns, for obvious reasons)
Tweaks loading code for the who knows how many-ith time

VickiMorris committed:
Auto-racking pump action

Pump-action guns now operate as they used to, and will rack themselves automatically.
Pump-action guns can be ctrl+clicked to pump and lock them open for breach loading, before closing manually.

VickiMorris committed:
NO_BOLT guns function like they used to (mostly)

NO_BOLT guns no longer function on a 3-step (open, unload, close) process.
NO_BOLT guns *still* require that the contents be dumped before loading new cartridges though.

VickiMorris committed:
Reimplements 3-Step NO_BOLT guns as new bolt type

Adds redesigned function of NO_BOLT as NB_BREAK
Reclassifies Revolvers, the Derringer, the Pipe Rifle, and Break Action Shotguns (Double Barrel, Improv, Bounty) as NB_BREAK

VickiMorris committed:
Makes guns not chamber on initialize

As it says, fucked up my if statement so it was always true.

VickiMorris committed:
Cleans up code slightly

NB_BREAK now uses var/bolt_locked to prevent firing (like every other gun)
Streamlines code for blow_up()

VickiMorris committed:
the shitcode was coming from INSIDE THE PR

Cleans up the logic of directly chambering rounds
Fixes loading PUMP bolt type guns (because I broke it)
Makes the code for directly chambering work for any gun that actually chambers rounds (if they have the proper variables set)

VickiMorris committed:
Adds Pipe Repeater (SMG), De-Pins Improvised Guns

Adds pipe repeater (bootleg WT-550)
Adds pipe repeater magazine (Upgradable with matter bins)
Piggybacks off of Uzi for pipe repeater sprites (CHANGE LATER)
Improvised weapons do not require a firing pin to function now (engineers make do)
Fixes OPEN_BOLT guns eating bullets when racked with a magazine installed

VickiMorris committed:
Pipe SMG Sprites

Adds sprites for pipe smg, and magazines (Tiers 1 to 5)
Renames sprites for pipe rifle to "piperifle" (instead of "pipegun")
Slightly tweaks how special mag overlays function, as it checks initial icon state instead of current.
Cleans up sprite code for bulldog magazines

VickiMorris committed:
fuck my stupid baka life

Adds glass-pack improv shells (lower damage, can embed though)
Adds improvised 9mm (same damage, high variance in accuracy)
Lowers Improv shell pellets, variance, damage. Adds AP & increases minimum range
Adds so many revolver sprites (To show open/closed cylinder)

VickiMorris committed:
(Most) Shotgun Bolt Overlays

VickiMorris committed:
Defines the musket, Commit to not lose progress

Adds musket ammo, adds most of the musket code & functionality
Renames pipe rifle internal mag to internal/piperifle instead of internal/pipegun
Fixes my doubledefine for hotload/softslug cartridges for the pipe rifle
Lowers the random spread for improv 9mm casings

VickiMorris committed:
shitcode compiles, will tear out later

On second thought, I'm just gonna make the musket itself a reagent container instead of adding an internal one and fucking around with shittons of snowflake code. It compiles in the current state but is a bullshit stupid implementation of how I want it to work bluh bluh bluh

VickiMorris committed:
Removes the shitcode, as promised :)

Removes the musket, it's a nightmare of coding and balancing and I'm not dealing with it.

VickiMorris committed:
Minor Tweaks

VickiMorris committed:
Merge branch 'master' into pipeGuns

VickiMorris committed:
Merge Upstream, Re-Remove Complex NO_BOLT

Removes NB_BREAK bolt type and resets all changed guns to NO_BOLT, the complexity isn't worth the headache.
Merges 250+ upstream commits as well.
Removes magazine from pipe rifle (only load into chamber, fun quirk of new system)
Reduce Capacity of Pipe SMG by 3 (6 to 18 capacity)
All improv guns now start empty (no free ammo for you!)
Makes pipe rifle slingable

VickiMorris committed:
Retypes glasspack shells

Just keeps it tidy, as it is a subtype of improv shell

VickiMorris committed:
Retype a41/paper to a41/improv

More accurate, ditching the paper cartridge hull idea.

VickiMorris committed:
All Ammo Boxes are now SMALL & Crafting Recipes

All ammo boxes and external magazines are now item weight SMALL instead of item weight TINY.
Crafting recipes for all improv guns, ammo types, and pipe SMG magazine.
New improv 9mm ammo pouch, 6 shots.

VickiMorris committed:
In-Hands for Improv Shottie & Pipe Rifle

In-hand and sawn in-hand sprites for improvised shotgun, and pipe rifle. Lever rifle uses lever action shotgun in-hands (that I don't even think the lever action shotgun uses).
Pipe SMG uses 'arg' in-hand sprite, similar to the WT-550 and NT-ARG.
Additional variable on improv shotguns to allow them to fire 'high-power' shells again (admin/mapper only for now)
Russian Shuttle gets reinforced improv sawn shotguns (to function as they do currently) and a hacksaw (to saw down their Mosins, if they wish.)
Adds a hacksaw (misc tool to saw down guns, very slow)

VickiMorris committed:
Fixes mapping issue?

Can't use true for maps, apparently, good to know for future.

VickiMorris committed:
Reverts some shotgun ammo changes

removes single-aught buckshot, reverts most changes to improv shells

VickiMorris committed:
Inches the lever rifle's lever forward a bit

VickiMorris committed:
Cleans up a few remaining touches

Fixes overlays for sniper.
Fixes audio for lever rifle.

VickiMorris committed:
Further increases range of improv shells

Range of range is now 5-10 instead of 3-8

VickiMorris committed:
Guts .41 ammo, retypes to a7.62 & .38

Removes a redundant ammo type, makes the pipe rifle fire a7.62 (Mosin Nagant), modifies the improv ammo to be a7.62 caliber, changes lever rifle to be loaded in .38 instead of .41
Tweaks (now) improv a7.62 ammo to be analogous to actual a7.62 ammo.
NICE_SHOT (admin only) trait allows racking PUMP and TWO_STEP guns one-handed
Slightly increases cost of improv 9mm ammo

VickiMorris committed:
Hacksaw available in hacked autolathes

Hacksaw can be printed from hacked autolathes for 4.5 sheets of iron, and 1.5 sheets of glass (base price).
Hacksaws can be used for surgery (x3 slower than circular saws), and for butchering (9 second butcher time, 80% efficacy, 10% bonus drop chance)
Moves hacksaws from ballistics.dm to surgery/tools.dm
Adds hacksaw noise :) (used when butchering only)
Original sound from larryblag on freesound.org
https://freesound.org/people/larryblag/sounds/333288/

VickiMorris committed:
Improv .38, .357

Makes ammo pouches a parent type
Adds improv .38 and .357 ammo
Lowers range of improvised shotgun pellets again (4-8 tiles)

VickiMorris committed:
Removes AP on improv 7.62

Heehoo fixes my issue too

VickiMorris committed:
Code Clean-Up Pass 1

Removes 7.62 Copper HP rounds (unneeded bloat)
Actually gives improv .38 the proper projectile
Removes the weird CALLBACK thing from BOLT_TYPE_PUMP guns, only uses locked-open state for special Ctrl+Click interaction.
Renames the 'pipe repeater' to 'pipe carbine'
Gives an examine text for the 'pipe rifle' and 'improvised shotgun' to let people know they can give it a shoulder sling from spare cabling.
Increases damage of improv .38
Reduces negative AP of improv .357
Marginally increases damage of 7.62 hotload, marginally reduces AP as well.
Adds minor recoil to the Mosin

VickiMorris committed:
Merge branch 'BeeStation:master' into pipeGuns

VickiMorris committed:
Merge remote-tracking branch 'upstream/master' into pipeGuns

VickiMorris committed:
Light Balance Tweaks

Bumps improv .357 up slightly (50 dmg -10 AP instead of 45 dmg -15 AP)
Lowers improv shotgun shell range again (3-8 instead of 4-8)
Adds improv 10mm (27 damage, 10 innate spread)

VickiMorris committed:
Fixes my runtime

VickiMorris committed:
Derringer's weight tweak

Derringer is now weapon weight 0.5, so unwielded spread is no longer hard defined.

VickiMorris committed:
Actually makes the change from the last commit.

VickiMorris committed:
Merge branch 'master' into pipeGuns

VickiMorris committed:
Update shotgun.dm

Runtime Fix Pt. 1

VickiMorris committed:
Update tools.dm

Runtime Fix Pt. 2

Rukofamicom commented on code/modules/research/designs/autolathe_designs.dm:
```diff
@@ -561,6 +561,14 @@
+	name = "Hacksaw"
+	id = "hacksaw"
+	build_type = AUTOLATHE
+	materials = list(/datum/material/iron=9000, /datum/material/glass=3000)
```
Why does the hacksaw take glass to produce?

VickiMorris replied:
IIRC I based the cost of the hacksaw off of the circular saw, tbh it does make more sense for it to be just pure iron instead of iron and glass. I'll tweak that to be 12000 iron instead of 9000 iron and 3000 glass when I get a chance.

VickiMorris replied:
Updated to be 12000 iron instead.

Rukofamicom replied:
👍

Rukofamicom commented on code/modules/projectiles/ammunition/ballistic/shotgun.dm:
```diff
@@ -5,26 +5,30 @@
 	desc = "A 12 gauge lead slug."
 	icon_state = "blshell"
 	caliber = "shotgun"
+	var/high_power = TRUE
```
This is purely a suggestion and not a forced requirement for merging:

Instead of makeshift/shitty shotguns not being allowed to use high_power rounds, would you consider making them have a chance to explode and harm the user if they try to use high powered rounds?

VickiMorris replied:
That'd probably be the better solution instead of a hard-ban on high-power shells. I'm not sure if I'd prefer it to be a flat percent chance (probably around 15%) each shot, or if high-power ammo 'damages' the barrel and builds up a malfunction chance (with it guaranteed to explode after 5 or so shots).

DarnTheMarn replied:
Apologies but I have to do the cliché and state my unsought opinion. I hate RNG based things like that, I myself would prefer if there was a set number of shots before it blew up. Its more fair that way.

Rukofamicom replied:
Strongly disagree - RNG absolutely deserves a place in combat within an RP oriented setting, especially as a limiter against some of the more powerful tools available. It would not deserve a place if this were a heavily pvp oriented server but this isn't the case.

Somewhere between 10% and 25% is a good rate - it's low enough that they can be used at least somewhat reliably, but high enough that malfunctions are a threat and prevent them from being a totally dominating factor.

DarnTheMarn replied:
I do think RNG in something like this where it only effects a person who gets greedy is far less egregious than say the tonfa baton PR where I said the same thing as that had the chance to just instantly sleep someone you used it on.

VickiMorris replied:
Just added a ~~bullshit RNG~~ stress mechanic to improvised shotguns. 10% base chance to explode (similar to det's revolver) that slowly goes up as high-power ammo is fired through it (+5% per shot). After enough strain is put on the gun (+30% / 6 shots), there's a slim chance (5%) for low-power ammo to cause a detonation as well. The reinforced variable is used to ignore this mechanic now.

Video is under testing evidence. Feels pretty alright in the little bit of testing I did with it tbh. There are pop-ups at certain stress thresholds, and it gives a vague idea of the current stress level in the examine prompt.

Rukofamicom replied:
It's almost eerie how that matches the compromise I was going to suggest number for number but got interrupted before I could finish typing it on lunch

DarnTheMarn replied:
Is this the extremely rare compromise that makes everyone equally happy instead of equally unhappy? I think it might be.

Rukofamicom reviewed (commented):
Just a couple of questions before approval

VickiMorris committed:
Update tools.dm

Modifies hacksaw to be 12000 iron instead of 9000 iron and 3000 glass

VickiMorris committed:
Update autolathe_designs.dm

Updated cost in autolathe for hacksaw.

VickiMorris committed:
Barrel Stress & Explosions

improv shotguns have a small chance to explode when firing high-power ammo, going up as more shots are put through the gun.

VickiMorris committed:
Vanishing Bullet Shrapnel

Shrapnel from bullets now has a chance to remain, rather than being an all or nothing roll. Should result in less clutter around the station (especially with glasspack rounds existing now)

VickiMorris committed:
Update shrapnel.dm

Slight bump to vanish chance on shrapnel

Rukofamicom commented on code/modules/projectiles/guns/ballistic/shotgun.dm:
```diff
@@ -282,6 +311,15 @@
+	else if (barrel_stress <= 25)
+		. += "The barrel seems to be warped mildly..."
+	else
+		. += "The barrel is cracked, and the warping is more severe!"
```
`The barrel is warped and cracked` would look better for the final message imo. The current wording assumes they have been examining it as it worsens, which will not always be the case. 

Additionally these thresholds don't match up with the warning messages very well right now: 
Thresholds should be `<10` and `<20` rather than `== 0` and `<= 25`.

VickiMorris replied:
Updated to have stuff line up. Barrel becomes warped at 10 stress (when message occurs), and cracked at 25 stress now.

After 2 shots, it's warped (20% misfire). After 5 shots, it's cracked (35% misfire).

VickiMorris committed:
updates barrel stress numbers

to_chat messages line up with examine message, threshold wise.

VickiMorris committed:
readability

easier to tell when thresholds are.

VickiMorris committed:
maybe I should READ first
